### ISSUE TITLE
USWDS - Enhancement: Choosing a date in the calendar on mobile opens keyboard unnecessarily 

### ISSUE THREAD

**Issue by ajanickiv** [uswds/uswds#5641]
### Describe the bug

Choosing a date in the calendar popup focuses the input and opens keyboard on mobile. This blocks any content below from being displayed and the user is forced to close the keyboard before continuing. 

### Steps to reproduce the bug

1. Go to the [date picker component on the USWDS site](https://designsystem.digital.gov/components/date-picker/) on a mobile device
2. Click the calendar button to open the calendar pop up
3. Choose a valid date
4. Calendar pop up will close, date will set in input, and focus changes to the input
5. Mobile keyboard will open
![mobile_datepicker_screenshot](https://github.com/uswds/uswds/assets/16595489/6e97e215-1615-45bd-ae68-e4a43c1e12a9)

### Expected Behavior

Date is set in the input but the focus stays on the calendar popup button. This should prevent the mobile keyboard from opening.

### Related code

In `packages/usa-date-picker/src/index.js` the focus is explicitly set to the input. A better user experience would be to place the focus on the calendar button.
![image](https://github.com/uswds/uswds/assets/16595489/177e1e18-4475-4c9b-b70c-e410a168f687)

### System setup

USWDS 3.7.0
Edge 119

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mejiaj** [comment]
Hey there @ajanickiv, thanks for creating this issue. 

I can confirm the keyboard pops up on Android browser (chrome). The component _does_ focus on the input and it seems logical to give users opportunity to modify entry.

I've moved this to `Not a bug` and set it as an enhancement since it would be a better experience to initially not show the keyboard.

**Comment by ajanickiv** [comment]
Going to disagree slightly here - since the user used the calendar popup to choose the date initially, if they wanted to modify that date, they would probably use the calendar popup again, not start typing to modify the date. This is why our users would prefer focus on the calendar popup button. If they wanted to type the date, the focus would have already been on the input and the keyboard visible.

Either way, thanks for looking at it.

**Comment by mejiaj** [comment]
Added `Needs discussion` to issue to discuss with team to get priority.

**TL;DR** Should we focus input or calendar button once a date has been selected.

**Comment by hallekfields** [comment]
Hi all! My team is affected by this issue as well. Is there any movement on this?
